### PR TITLE
Cleaning the contact_map

### PR DESCRIPTION
**Pull request by hengma1001**
1. Moved the contact_size definition before the h5 file path where the variable is used 
2. Saved the whole contact_matrices to h5 file instead of doing it line by line 
3. Moved file saving out of trajectory loop

### CODE REVIEW

File: molecules/sim/md_analysis/contact_maps.py [hunk 1]
```diff
@@ -15,25 +15,25 @@ def contact_maps_from_traj(xyz_file, traj_file, cm_table_file, contact_cutoff=8.
...
 
-        if savepath and filename:
-            cm_table.append(reshaped_matrix)
-            outfile.close()
+    if savepath and filename:
+        cm_table.append(reshaped_matrix)
+        outfile.close() 
+        return outpath
 
     return contact_matrices
```
Comment (yngtodd) [inline molecules/sim/md_analysis/contact_maps.py:32]: What would happen if you want to both write to disk and return `contact_matrices` for other parts of the program?

File: molecules/utils/matrix_op.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+def tria_to_full(cm0): 
+    num_res = int(np.ceil((len(cm0) * 2) ** 0.5))
+    iu1 = np.triu_indices(num_res, 1)
+
+    cm_full = np.zeros((num_res, num_res))
+    cm_full[iu1] = cm0 
+    cm_full.T[iu1] = cm0 
+    np.fill_diagonal(cm_full, 1)
+    
+    return cm_full
```
Comment (yngtodd) [inline molecules/utils/matrix_op.py:12]: Nice. You are awesome.

**Comment by hengma1001** [comment]
Let me add that also. Lol


On Oct 25, 2018, at 4:19 PM, Todd Young <notifications@github.com<mailto:notifications@github.com>> wrote:


@yngtodd commented on this pull request.

________________________________

In molecules/utils/matrix_op.py<https://github.com/yngtodd/molecules/pull/6#discussion_r228320169>:

> @@ -0,0 +1,12 @@
+import numpy as np
+
+def tria_to_full(cm0):
+    num_res = int(np.ceil((len(cm0) * 2) ** 0.5))
+    iu1 = np.triu_indices(num_res, 1)
+
+    cm_full = np.zeros((num_res, num_res))
+    cm_full[iu1] = cm0
+    cm_full.T[iu1] = cm0
+    np.fill_diagonal(cm_full, 1)
+
+    return cm_full


Nice. You are awesome.

File: molecules/sim/md_analysis/contact_maps.py (hunk 1 above)
Comment (yngtodd) [reply molecules/sim/md_analysis/contact_maps.py:32]: Cool, that looks good! 👍

**Comment by yngtodd** [comment]
Cool! I'm going to go ahead and merge this one, and we can open up a new pull request if there is anything else.

**Pull request merged**